integrations/coinbase-integration/src/main/java/org/dash/wallet/integration/coinbase_integration/ui/CoinbaseBuyDashOrderReviewFragment.kt

```diff
@@ -159,6 +165,14 @@ class CoinbaseBuyDashOrderReviewFragment : Fragment(R.layout.fragment_coinbase_b
             it.updateOrderReviewUI()
             countDownTimer.start()
         }
+
+        parentFragmentManager.beginTransaction()
+            .replace(R.id.network_status_container, NetworkUnavailableFragment.newInstance())
+            .commit()
+
+        amountViewModel.isDeviceConnectedToInternet.observe(viewLifecycleOwner){ hasInternet ->
+            setNetworkState(hasInternet)
+        }
```

Comment:
I'm not sure that it is the responsibility of the `EnterAmountViewModel` to track the connectivity state. Its responsibility is only to enter an amount, so we're breaking the single responsibility principle here. Perhaps it's better to derive `CoinbaseBuyDashOrderReviewViewModel` from the `ConnectivityViewModel`?